src/main/java/us/kbase/auth2/providers/OrcIDIdentityProviderFactory.java

```diff
@@ -228,6 +233,58 @@ private OrcIDAccessTokenResponse(
 				this.accessToken = accessToken.trim();
 				this.fullName = fullName == null ? null : fullName.trim();
 				this.orcID = orcID.trim();
+				this.idToken = idToken == null ? null : idToken.trim();
+				this.mfaAuthenticated = parseAmrClaim(this.idToken);
+			}
+			
+			/**
+			 * Parses the Authentication Method Reference (AMR) claim from an OpenID Connect ID token
+			 * to determine if multi-factor authentication was used.
+			 * 
+			 * @param idToken the JWT ID token from ORCID
+			 * @return true if MFA was used, false if password only, null if unknown or parsing failed
+			 */
+			private Boolean parseAmrClaim(final String idToken) {
+				if (idToken == null || idToken.trim().isEmpty()) {
+					return null;
+				}
+				
+				try {
+					// JWT format: header.payload.signature
+					final String[] parts = idToken.split("\\.");
```

Comment:
is this copied from the Google provider? I seem to remember something like this. I'm pretty sure I wasn't happy about how that was implemented; it'd be better to use a library if possible

Reply:
Modified from some Orcid openid examples; will look for a library

Reply:
I would look, but timebox yourself to an hour two is my advice. I remember looking but not finding one that was reasonably simple to use

Reply:
Seems most libraries that do this are pretty heavy (eg, auth0) and focus on creation and validation of tokens. Looking online, manual parsing seems to be common in Java for just reading the claims. But, [auth0/java-jwt](https://github.com/auth0/java-jwt) is an option. I think the hand-rolled code is probably fine here, unless we want to start verifying the signature, or something.